src/clj/rems/db/entitlements.clj

```diff
@@ -92,21 +115,24 @@
                 :outbox/entitlement-post {:action action
                                           :entitlements entitlements}}))
 
-(defn- grant-entitlements! [application-id user-id resource-ids]
+(defn- grant-entitlements! [application-id user-id resource-ids actor]
   (log/info "granting entitlements on application" application-id "to" user-id "resources" resource-ids)
   (doseq [resource-id (sort resource-ids)]
     (db/add-entitlement! {:application application-id
                           :user user-id
-                          :resource resource-id})
+                          :resource resource-id
+                          :approvedby actor})
     ;; TODO could generate only one outbox entry per application. Currently one per user-resource pair.
-    (add-to-outbox! :add (db/get-entitlements {:application application-id :user user-id :resource resource-id}))))
+    (add-to-outbox! :add (db/get-entitlements {:application application-id :user user-id :resource resource-id}))
+    (add-to-outbox! :ga4gh (db/get-entitlements {:application application-id :user user-id :resource resource-id}))))
 
-(defn- revoke-entitlements! [application-id user-id resource-ids]
+(defn- revoke-entitlements! [application-id user-id resource-ids actor]
   (log/info "revoking entitlements on application" application-id "to" user-id "resources" resource-ids)
   (doseq [resource-id (sort resource-ids)]
     (db/end-entitlements! {:application application-id
                            :user user-id
-                           :resource resource-id})
+                           :resource resource-id
+                           :revokedby actor})
     (add-to-outbox! :remove (db/get-entitlements {:application application-id :user user-id :resource resource-id}))))
```

Comment:
revoked aren't posted to ga4gh? should they?

Reply:
Not at this state apparently, asked @jaakkocsc :)